node-js-basics-101/node_modules/socket.io-adapter/index.js

```diff
@@ -0,0 +1,263 @@
+
+/**
+ * Module dependencies.
+ */
+
+var Emitter = require('events').EventEmitter;
+
+/**
+ * Module exports.
+ */
+
+module.exports = Adapter;
+
+/**
+ * Memory adapter constructor.
+ *
+ * @param {Namespace} nsp
+ * @api public
+ */
+
+function Adapter(nsp){
+  this.nsp = nsp;
+  this.rooms = {};
+  this.sids = {};
+  this.encoder = nsp.server.encoder;
+}
+
+/**
+ * Inherits from `EventEmitter`.
+ */
+
+Adapter.prototype.__proto__ = Emitter.prototype;
```

Comment:
__[jshint]__
* The '__proto__' property is deprecated.

<!-- Comment by [SideCI](https://sideci.com) -->